packages/algorand-js/src/runtime/runtime.ts

```diff
@@ -0,0 +1,215 @@
+/* eslint sonarjs/no-duplicate-string: 0 */
+/* eslint sonarjs/no-small-switch: 0 */
+import { mkTransaction } from "algob";
+import { ExecParams, TransactionType } from "algob/src/types";
+import { getProgram } from "algob/test/helpers/fs";
+import { assignGroupID, SSCParams, SSCStateSchema } from "algosdk";
+
+import { mockSuggestedParams } from "../../build/test/mocks/txn";
+import { TealError } from "../errors/errors";
+import { ERRORS } from "../errors/errors-list";
+import { Interpreter } from "../index";
+import { BIGINT0 } from "../interpreter/opcode-list";
+import { checkIndexBound, compareArray } from "../lib/compare";
+import { convertToString } from "../lib/parsing";
+import { assertValidSchema, getKeyValPair } from "../lib/stateful";
+import type { Context, SdkAccount, StackElem, Txn } from "../types";
+
+export class Runtime {
+  interpreter: Interpreter;
```

Comment:
we have `interpreter` in `runtime` and 
`runtime` in `interpreter`. I don't think it is good to have cycle. Linear Flow is easy to understand for developers and users.
(just a thought)

Reply:
I agree. Interpreter needs runtime to access runtime state functions, and Runtime should take interpreter (as we are executing .teal code). 
One idea is to represent `interpreter` as a child class of `Runtime` and interpreter can access runtime functions using `super`.
Did you have something in mind?

Reply:
One idea that  i have we can put these functions in `runtime.ts` but out of runtime class, because these functions are only using `this.interpreter.ctx`. we can export them and use them directly from interpreter. But I am not sure if this is a good way

Reply:
another way is to remove `interpreter` from runtime, create a new object in executeTx function only.
Update globalState and other functions to work from `ctx` passed(as i can see they require this.interpreter.ctx).

Reply:
> One idea that i have we can put these functions in runtime.ts but out of runtime class, because these functions are only using this.interpreter.ctx. we can export them and use them directly from interpreter. But I am not sure if this is a good way

Agreed, not a good way.

Reply:
> another way is to remove `interpreter` from runtime, create a new object in executeTx function only.
Update globalState and other functions to work from `ctx` passed(as i can see they require this.interpreter.ctx only).

Yes, i think we can:
- remove interpreter from runtime (constructor), and declare it directly in `runtime.executeTx()`. Or
- Make `interpreter` class child of `Runtime`, and interpreter can access runtime functions using `this.super...`.

But in both approaches, it means we are using `interpreter` internally and user is only exposed to `Runtime`. @robert-zaremba  what do you think?

Reply:
was this resolved @ratik21 ?

Reply:
Yes, Ratik moved interpreter creation to `runtime.executeTx`: https://github.com/scale-it/algorand-builder/pull/169/files#diff-614a0db207a77ce83f6fd3b00f9c0e43c40ec8f7d010aa2b15cbfe5d9439c438